modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/AbstractWorkflowOperationHandler.java

```diff
@@ -360,6 +378,160 @@ protected String getConfig(WorkflowOperationInstance woi, String key) throws Wor
     return Opt.nul(woi.getConfiguration(key)).flatMap(Strings.trimToNone);
   }
 
+  /**
+   * Returns a ConfiguredTagsAndFlavors instance, which includes all specified source/target tags and flavors if they are valid
+   * Lists can be empty, if no values were specified! This is to enable WOHs to individually check if a given tag/flavor was set.
+   * @param srcTags none, one or many
+   * @param srcFlavors none, one or many
+   * @param targetFlavors none, one or many
+   * @param targetTags none, one or many
+   * @return ConfiguredTagsAndFlavors object including lists for the configured tags/flavors
+   */
+  protected ConfiguredTagsAndFlavors getTagsAndFlavors(WorkflowInstance wi, Configuration srcTags, Configuration srcFlavors, Configuration targetTags, Configuration targetFlavors) throws WorkflowOperationException {
+    WorkflowOperationInstance woi = wi.getCurrentOperation();
```

Comment:
```suggestion
  protected ConfiguredTagsAndFlavors getTagsAndFlavors(WorkflowInstance workflow, Configuration srcTags, Configuration srcFlavors, Configuration targetTags, Configuration targetFlavors) throws WorkflowOperationException {
    WorkflowOperationInstance operation = wi.getCurrentOperation();
```

Good names are good :smirk_cat: